x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.tsx

```diff
@@ -104,11 +106,13 @@ export const LinksMenuUI = (props: LinksMenuProps) => {
   const { getDataViewById, getDataViewIdFromName } = useMlIndexUtils();
 
   const job = useMemo(() => {
+    if (props.selectedJob !== undefined) return props.selectedJob;
     return mlJobService.getJob(props.anomaly.jobId);
-  }, [props.anomaly.jobId]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
From my testing, this rule suppression isn't needed and instead we can just use `[props.anomaly.jobId, props.selectedJob]` in the dependency list

Reply:
Updated in 53b573ff6d8d57acfe1ca468df74f5a8d63354a3